lib/src/rules/prefer_sizedbox_over_container.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+import '../util/flutter_utils.dart';
+
+const _desc = r'Prefer SizedBox over Container.';
+
+const _details = r'''Prefer SizedBox over Container for introducing blank space.
+
+A `Container` is a heavier Widget than a `SizedBox`, and as bonus, `SizedBox` 
+has a `const` constructor.
+
+**BAD:**
+```
+Widget buildRow() {
+  return Row(
+    children: <Widget>[
+      const MyLogo(),
+      Container(width: 4),
+      const Expanded(
+        child: Text('...'),
+      ),
+    ],
+  );
+}
+```
+
+**GOOD:**
+```
+Widget buildRow() {
+  return Row(
+    children: const <Widget>[
+      MyLogo(),
+      SizedBox(width: 4),
+      Expanded(
+        child: Text('...'),
+      ),
+    ],
+  );
+}
+```
+''';
+
+class PreferSizedBoxOverContainer extends LintRule implements NodeLintRule {
+  PreferSizedBoxOverContainer()
+      : super(
+            name: 'prefer_sizedbox_over_container',
```

Comment:
We have decided to move away from the "always", "prefer", etc. terminology, at least for rules related to the style guide (which this isn't) because those designations tend to change over time and several of our rules are now misnamed. I don't know whether we want to apply the same reasoning to non-style-guide rules. @pq Thoughts?

Also, a minor nit, I think "sizedbox" wants to be "sized_box".

Other than those questions, lgtm.

Reply:
How about naming it `sized_box_for_whitespace`?

Reply:
Brian is right.  Where possible, it's nice to avoid names that suggest severities.  To that end, `sized_box_for_whitespace` sounds great to me!